server/pbenchinacan/run-pbench-in-a-can

```diff
@@ -1,142 +1,67 @@
-#   8080 - HTTP web server (and Pbench Dashboard)
 
-# The default container registry to use, and the organization name in that
-# registry.
+# The default container registry to use with the organization name in that
+# registry.  E.g., `quay.io/pbench`.  We export it (and the server image name)
+# for the subordinate scripts to use.
 export PB_CONTAINER_REG=${PB_CONTAINER_REG:-$(<${HOME}/.config/pbench/ci_registry.name)}
+server_image_name=${PB_SERVER_IMAGE_NAME:-"pbench-server"}
+server_image_tag=${PB_SERVER_IMAGE_TAG:-"$(< jenkins/branch.name)"}
+export PB_SERVER_IMAGE=${PB_CONTAINER_REG}/${pb_server_image_name}:${pb_server_image_tag}
 
 # Default pull policy for all images is to pull the image if the registry has a
-# newer version.
+# *different* version (despite the name...different is assumed to be "newer").
 export PB_COMMON_IMAGE_PULL_POLICY=${PB_COMMON_IMAGE_PULL_POLICY:-"newer"}
-
-# Each container has its own name and tag for the image, as well as a pull
-# policy.
-export PB_ELASTICSEARCH_IMAGE_NAME=${PB_ELASTICSEARCH_IMAGE_NAME:-"elasticsearch"}
-export PB_ELASTICSEARCH_IMAGE_TAG=${PB_ELASTICSEARCH_IMAGE_TAG:-"7.13.3"}
-export PB_ELASTICSEARCH_IMAGE="${PB_CONTAINER_REG}/${PB_ELASTICSEARCH_IMAGE_NAME}:${PB_ELASTICSEARCH_IMAGE_TAG}"
-export PB_ELASTICSEARCH_IMAGE_PULL_POLICY="${PB_ELASTICSEARCH_IMAGE_PULL_POLICY:-${PB_COMMON_IMAGE_PULL_POLICY}}"
-
-export PB_POSTGRESQL_IMAGE_NAME=${PB_POSTGRESQL_IMAGE_NAME:-"postgresql-13"}
-export PB_POSTGRESQL_IMAGE_TAG=${PB_POSTGRESQL_IMAGE_TAG:-"latest"}
-export PB_POSTGRESQL_IMAGE="${PB_CONTAINER_REG}/${PB_POSTGRESQL_IMAGE_NAME}:${PB_POSTGRESQL_IMAGE_TAG}"
-export PB_POSTGRESQL_IMAGE_PULL_POLICY="${PB_POSTGRESQL_IMAGE_PULL_POLICY:-${PB_COMMON_IMAGE_PULL_POLICY}}"
-
-export PB_KEYCLOAK_IMAGE_NAME=${PB_KEYCLOAK_IMAGE_NAME:-"pbenchinacan-keycloak"}
-export PB_KEYCLOAK_IMAGE_TAG=${PB_KEYCLOAK_IMAGE_TAG:-"20.0.3"}
-export PB_KEYCLOAK_IMAGE="${PB_CONTAINER_REG}/${PB_KEYCLOAK_IMAGE_NAME}:${PB_KEYCLOAK_IMAGE_TAG}"
-export PB_KEYCLOAK_IMAGE_PULL_POLICY="${PB_KEYCLOAK_IMAGE_PULL_POLICY:-${PB_COMMON_IMAGE_PULL_POLICY}}"
-
-export PB_SERVER_IMAGE_NAME=${PB_SERVER_IMAGE_NAME:-"pbench-server"}
-export PB_SERVER_IMAGE_TAG=${PB_SERVER_IMAGE_TAG:-"$(cat jenkins/branch.name)"}
-export PB_SERVER_IMAGE="${PB_CONTAINER_REG}/${PB_SERVER_IMAGE_NAME}:${PB_SERVER_IMAGE_TAG}"
 export PB_SERVER_IMAGE_PULL_POLICY="${PB_SERVER_IMAGE_PULL_POLICY:-${PB_COMMON_IMAGE_PULL_POLICY}}"
 
-# The overall pod name is always derived from the pbench server image tag.
-export PB_POD_NAME=${PB_POD_NAME:-"pbench-in-a-can_${PB_SERVER_IMAGE_TAG}"}
-
 # Directory to use for the fully built dashboard code.
 export PB_DASHBOARD_DIR="${PB_DASHBOARD_DIR:-${PWD}/dashboard/build/}"
 
-podman pod create --name ${PB_POD_NAME} -p 9200:9200 -p 5432:5432 -p 8090:8090
+export KEYCLOAK_REALM=${KEYCLOAK_REALM:-"pbench-server"}
+export KEYCLOAK_CLIENT=${KEYCLOAK_CLIENT:-"pbench-server-client"}
```

Comment:
@npalaska, @portante, are these the correct/appropriate values?  (They'll be used both to set the Pbench Server config file and to configure the canned Keycloak server.)

Reply:
These will need to be configurable, since the Integration Lab deployment may change them.

Reply:
They are configurable -- if either `KEYCLOAK_*` variable is defined in the environment, then the values provided here will be ignored.

Reply:
I would say we keep the default client `pbench-dashboard` but that's a deployment detail I wouldn't worry about. Like you said we could configure `KEYCLOAK_*` variable to set them.